pkgs/applications/networking/cluster/hadoop/default.nix

```diff
@@ -1,23 +1,45 @@
-{ lib, stdenv, fetchurl, makeWrapper, autoPatchelfHook
-, jdk8_headless, jdk11_headless
-, bash, coreutils, which
-, bzip2, cyrus_sasl , protobuf3_7, snappy, zlib, zstd
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, autoPatchelfHook
+, jdk8_headless
+, jdk11_headless
+, bash
+, coreutils
+, which
+, bzip2
+, cyrus_sasl
+, protobuf3_7
+, snappy
+, zlib
+, zstd
 , openssl
 }:
 
 with lib;
 
 let
   common = { pname, version, untarDir ? "${pname}-${version}", sha256, jdk, openssl, nativeLibs ? [ ], libPatches ? "" }:
+    let
+      platformUrlSuffix =
+        if stdenv.isx86_64 && (stdenv.isLinux || stdenv.isDarwin)
+        then ""
+        else
+          if stdenv.isAarch64 && (stdenv.isLinux || stdenv.isDarwin)
+          then "-aarch64"
+          else throw "Hadoop does not currently support ${stdenv.system}.";
+    in
```

Comment:
generally we don't bother with doing platform radix, as that's captured in the `.drv`

Reply:
```
[13:26:04] jon@nixos ~/projects/nixpkgs (staging-next)
$ nix-instantiate -A hadoop
warning: you did not specify '--add-root'; the result might be removed by the garbage collector
/nix/store/smsf7vdmscgj1fqdvc7b14fw186pn7kn-hadoop-3.3.1.drv
[13:47:28] jon@nixos ~/projects/nixpkgs (staging-next)
$ nix show-derivation /nix/store/smsf7vdmscgj1fqdvc7b14fw186pn7kn-hadoop-3.3.1.drv | grep system
    "system": "x86_64-linux",
      "system": "x86_64-linux",
````

Reply:
Does <https://github.com/NixOS/nixpkgs/pull/158613/commits/a24835e6c6cb5e6a96d5fd0ec625abb64894cfb0> handle it better?